### PR TITLE
Created project and added styles, components and basic settings

### PR DESCRIPTION
**Pull request by Ilya-Pargoo**
Added Layout, common with footer, header, buttons, icons, menus. Updated basic tailwind.config, .eslintrc, postcss.config

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  },
+  "dependencies": {
+    "@fontsource/open-sans": "^5.1.0",
+    "next": "15.0.3",
+    "react": "19.0.0-rc-66855b96-20241106",
+    "react-dom": "19.0.0-rc-66855b96-20241106"
+  },
+  "devDependencies": {
+    "@types/node": "^20",
+    "@types/react": "^18",
```
Comment (Ilya-Valasiuk) [inline package.json:20]: ?

Comment (Ilya-Valasiuk) [reply package.json:20]: ^

File: src/components/Layout/Layout.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { Footer } from "../common/footer"
+import { Header } from "../common/header"
+
+export const Layout = ({ children }: { children: React.ReactNode }) =>
```
Comment (Ilya-Valasiuk) [inline src/components/Layout/Layout.tsx]: folder name and location

Comment (Ilya-Valasiuk) [reply src/components/Layout/Layout.tsx]: what is children type?

File: src/components/common/buttons/BurgerMenuButton/BurgerMenuButton.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { FC } from "react";
+import { BurgerMenuButtonProps } from "./types";
+
+export const BurgerMenuButton: FC<BurgerMenuButtonProps> = ({ onClick, isOpen }) =>
+{
```
Comment (Ilya-Valasiuk) [inline src/components/common/buttons/BurgerMenuButton/BurgerMenuButton.tsx]: ??

File: src/components/common/buttons/BurgerMenuButton/BurgerMenuButton.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    return (
+        <button
+            className="md:hidden flex items-center focus:outline-none"
+            onClick={onClick}
+        >
+            <svg
```
Comment (Ilya-Valasiuk) [inline src/components/common/buttons/BurgerMenuButton/BurgerMenuButton.tsx]: ?

File: src/components/common/buttons/BurgerMenuButton/BurgerMenuButton.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import { FC } from "react";
+import { BurgerMenuButtonProps } from "./types";
+
+export const BurgerMenuButton: FC<BurgerMenuButtonProps> = ({ onClick, isOpen }) =>
+{
+    return (
+        <button
+            className="md:hidden flex items-center focus:outline-none"
```
Comment (Ilya-Valasiuk) [inline src/components/common/buttons/BurgerMenuButton/BurgerMenuButton.tsx]: styleguide

Comment (Ilya-Valasiuk) [reply src/components/common/buttons/BurgerMenuButton/BurgerMenuButton.tsx]: https://drive.google.com/drive/folders/1WmL9ZSnM0yoYJtSMVwqDVBdEBKeVxJVt?ths=true

File: src/components/common/buttons/BurgerMenuButton/types.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+export interface BurgerMenuButtonProps
```
Comment (Ilya-Valasiuk) [inline src/components/common/buttons/BurgerMenuButton/types.ts]: no interfaces. why it's in new folder?

File: src/components/common/footer/Footer.tsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+import { navItems } from "./constants";
+import { TwitterIcon } from "../icons/TwitterIcon";
+import { FacebookIcon } from "../icons/FacebookIcon";
+import { InstagramIcon } from "../icons/InstagramIcon";
+
+export const Footer: FC = () =>
+{
+    return (
+        <footer className="bg-white">
+            <div className="container py-5 border-t" style={{ borderTop: '1px solid var(--separator)' }}>
```
Comment (Ilya-Valasiuk) [inline src/components/common/footer/Footer.tsx]: what is style for?

File: src/components/common/footer/Footer.tsx [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+                <div className="flex flex-col gap-y-8 text-center md:flex-row md:justify-between md:items-center md:space-y-0 md:text-left">
+                    <p className="footer-link pointer-events-none order-1 md:order-none">© 2024 Best News</p>
+                    <NavList
+                        navClass={""}
+                        navItemListClass={"flex flex-col space-y-5 md:flex-row md:space-y-0 md:space-x-8 justify-center"}
+                        navItemClass={"footer-link"}
+                        navItems={navItems}
+                    />
+                    <div className="flex justify-center space-x-8 md:space-x-4">
+                        <TwitterIcon link={"https://twitter.com"} />
```
Comment (Ilya-Valasiuk) [inline src/components/common/footer/Footer.tsx]: what if we will need to use twitter just as icon in project

File: src/components/common/footer/constants.ts [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import { NavItem } from "../header/types";
```
Comment (Ilya-Valasiuk) [inline src/components/common/footer/constants.ts]: it's not possible to take types from another component unless it's children component

File: src/components/common/header/Header.tsx [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+import { MainLogoIcon } from "../icons/MainLogoIcon"
+import { navItems } from "./constants"
+import { BurgerMenuButton } from "../buttons/BurgerMenuButton"
+import { NavList } from "../menus/NavList"
+
+export const Header: FC = () =>
+{
+    const [isMobileMenuOpen, setMobileMenuOpen] = useState(false);
+
+    const toggleMobileMenu = useCallback(() => { setMobileMenuOpen((prev) => !prev); }, []);
```
Comment (Ilya-Valasiuk) [inline src/components/common/header/Header.tsx]: why useCallback here?

File: src/components/common/header/Header.tsx [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+
+    return (
+        <header className="shadow-custom-header content-center sticky top-0 z-50">
+            <div className="container h-full flex items-center justify-between  min-h-[60px]">
```
Comment (Ilya-Valasiuk) [inline src/components/common/header/Header.tsx]: 60 / 4 = 15. I guess it's predefined values

File: src/components/common/header/Header.tsx [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+                <NavList
+                    navClass={"hidden md:block"}
+                    navItemListClass={"flex space-x-6"}
+                    navItemClass={"header-link"}
+                    currentLinkClass="active"
+                    navItems={navItems}
+                />
+                <BurgerMenuButton onClick={toggleMobileMenu} isOpen={isMobileMenuOpen} />
+            </div>
+            {isMobileMenuOpen &&
```
Comment (Ilya-Valasiuk) [inline src/components/common/header/Header.tsx]: https://dev.to/ubahthebuilder/stop-conditional-rendering-in-react-without-knowing-this-and-ternary-operator-388

File: src/components/common/menus/NavList/types.ts [hunk 13]
```diff
@@ -0,0 +1,10 @@
+import { NavItem } from "../../header/types";
+
+export interface NavListProps
+{
+    navItems: NavItem[];
+    navItemClass: string;
+    navItemListClass: string;
+    navClass: string;
+    currentLinkClass?: string;
```
Comment (Ilya-Valasiuk) [inline src/components/common/menus/NavList/types.ts]: what's the reason to do this?

File: src/components/common/footer/Footer.tsx [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+
+export const Footer: FC = () =>
+{
+    return (
+        <footer className="bg-white">
+            <div className="container py-5 border-t" style={{ borderTop: '1px solid var(--separator)' }}>
+                <div className="flex flex-col gap-y-8 text-center md:flex-row md:justify-between md:items-center md:space-y-0 md:text-left">
+                    <p className="footer-link pointer-events-none order-1 md:order-none">© 2024 Best News</p>
+                    <NavList
+                        navClass={""}
```
Comment (Ilya-Valasiuk) [inline src/components/common/footer/Footer.tsx]: why not pass props in that case?

File: src/components/common/footer/Footer.tsx [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+                        navItemListClass={"flex flex-col space-y-5 md:flex-row md:space-y-0 md:space-x-8 justify-center"}
+                        navItemClass={"footer-link"}
```
Comment (Ilya-Valasiuk) [inline src/components/common/footer/Footer.tsx]: we using tailwind to get rid of css in general

File: src/components/common/icons/FacebookIcon/FacebookIcon.tsx [hunk 16]
```diff
@@ -0,0 +1,13 @@
+import Link from "next/link";
+import { FC } from "react";
+
+export const FacebookIcon: FC<{ link: string }> = ({ link }) =>
+{
+    return (
+        <Link href={link} className="group flex items-center" target="_blank" rel="noopener noreferrer">
+            <svg viewBox="0 0 18 18" fill="none" xmlns="http://www.w3.org/2000/svg" className="fill-subtext group-hover:fill-text transition-all duration-300 w-[40px] md:w-[24px]">
```
Comment (Ilya-Valasiuk) [inline src/components/common/icons/FacebookIcon/FacebookIcon.tsx]: 40 - 10

24 - 6

File: src/components/common/icons/FacebookIcon/FacebookIcon.tsx [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+                <path fillRule="evenodd" clip-rule="evenodd" d="M0 9.05025C0 13.5248 3.24975 17.2455 7.5 18V11.4998H5.25V9H7.5V6.99975C7.5 4.74975 8.94975 3.50025 11.0002 3.50025C11.6497 3.50025 12.3503 3.6 12.9998 3.69975V6H11.85C10.7498 6 10.5 6.54975 10.5 7.25025V9H12.9L12.5002 11.4998H10.5V18C14.7502 17.2455 18 13.5255 18 9.05025C18 4.0725 13.95 0 9 0C4.05 0 0 4.0725 0 9.05025Z" />
```
Comment (Ilya-Valasiuk) [inline src/components/common/icons/FacebookIcon/FacebookIcon.tsx]: you just need to use `currentColor` in svg

File: src/components/common/menus/NavList/NavList.tsx [hunk 18]
```diff
@@ -0,0 +1,26 @@
+import Link from "next/link";
+import { useRouter } from "next/router";
+import { FC } from "react"
+import { NavListProps } from "./types";
+
+export const NavList: FC<NavListProps> = (props) =>
```
Comment (Ilya-Valasiuk) [inline src/components/common/menus/NavList/NavList.tsx]: menus ?

File: src/pages/api/hello.ts [hunk 19]
```diff
@@ -0,0 +1,13 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
+import type { NextApiRequest, NextApiResponse } from "next";
+
+type Data = {
+  name: string;
+};
+
+export default function handler(
```
Comment (Ilya-Valasiuk) [inline src/pages/api/hello.ts]: ?

File: src/pages/index.tsx [hunk 20]
```diff
@@ -0,0 +1,6 @@
+export default function Home()
+{
+  return (
+    <div className="h-screen"></div>
```
Comment (Ilya-Valasiuk) [inline src/pages/index.tsx]: h-screen or min-h-screen?

File: src/styles/globals.css [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+:root {
+  /* Fonts */
+  --font-family: "Open Sans", sans-serif;
+
+  /* Colors */
+  --background: #fbfbfb;
+  --text: #161616;
+  --subtext: #7a7a7a;
+  --main: #e33a6d;
+  --separator: #d1d1d1;
```
Comment (Ilya-Valasiuk) [inline src/styles/globals.css]: define colors in tailwind config https://tailwindcss.com/docs/theme#customizing-the-default-theme

File: src/styles/globals.css [hunk 22]
```diff
@@ -0,0 +1,39 @@
...
+
+  /* Shadows */
+  --custom-header-shadow: 0 2px 10px 0 rgba(22, 22, 22, 0.1);
```
Comment (Ilya-Valasiuk) [inline src/styles/globals.css]: the same

File: package.json [hunk 23]
```diff
@@ -0,0 +1,29 @@
...
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "@fontsource/open-sans": "^5.1.0",
```
Comment (Ilya-Valasiuk) [inline package.json]: ?

File: src/components/common/buttons/BurgerMenuButton/BurgerMenuButton.tsx [hunk 24]
```diff
@@ -0,0 +1,27 @@
+import { FC } from "react";
```
Comment (Ilya-Valasiuk) [inline src/components/common/buttons/BurgerMenuButton/BurgerMenuButton.tsx]: burger-menu-button/BurgerMenuButton.tsx

File: src/components/Layout/Layout.tsx (hunk 2 above)
Comment (Ilya-Valasiuk) [reply src/components/Layout/Layout.tsx]:
```
export const Layout: React.PropsWithChildren = (children) =>
```

File: src/components/common/buttons/BurgerMenuButton/BurgerMenuButton.tsx (hunk 3 above)
Comment (Ilya-Valasiuk) [inline src/components/common/buttons/BurgerMenuButton/BurgerMenuButton.tsx]:
```
type Props = {
}

export const [NameComponent]: React.FC<Props> = ({  }) => {

```

File: src/components/common/buttons/BurgerMenuButton/BurgerMenuButton.tsx (hunk 5 above)
Comment (Ilya-Valasiuk) [reply src/components/common/buttons/BurgerMenuButton/BurgerMenuButton.tsx]:
```
className={classNames("flex items-center focus:outline-none", "md:hidden", "lg:flex", "2xl:hidden")}
```

```
className={classNames("flex items-center", "focus:outline-none", "md:hidden", "lg:flex", "2xl:hidden")}
```

File: src/components/common/buttons/BurgerMenuButton/BurgerMenuButton.tsx (hunk 4 above)
Comment (Ilya-Valasiuk) [reply src/components/common/buttons/BurgerMenuButton/BurgerMenuButton.tsx]: https://react-svgr.com/docs/next/

File: src/components/common/buttons/BurgerMenuButton/types.ts [hunk 25]
```diff
@@ -0,0 +1,5 @@
...
+{
+    onClick: () => void;
+    isOpen: boolean;
```
Comment (Ilya-Valasiuk) [inline src/components/common/buttons/BurgerMenuButton/types.ts]: order:

```
title
description
titleEn?: string;
isOpen: boolean;
onBlur: () => void;
onSelect: () => void;

```

File: tailwind.config.ts [hunk 26]
```diff
@@ -0,0 +1,34 @@
+import type { Config } from "tailwindcss";
+
+export default {
+  content: [
+    "./src/pages/**/*.{js,ts,jsx,tsx,mdx}",
```
Comment (Ilya-Valasiuk) [inline tailwind.config.ts]: content: ['./src/**/**/*.{js,jsx,ts,tsx}'],

File: src/components/common/menus/NavList/NavList.tsx [hunk 27]
```diff
@@ -0,0 +1,26 @@
+import Link from "next/link";
+import { useRouter } from "next/router";
+import { FC } from "react"
+import { NavListProps } from "./types";
+
```
Comment (Ilya-Valasiuk) [inline src/components/common/menus/NavList/NavList.tsx]: type Props = {
className?: string
linkClassName
}

File: src/components/common/menus/NavList/NavList.tsx [hunk 28]
```diff
@@ -0,0 +1,26 @@
...
+export const NavList: FC<NavListProps> = (props) =>
+{
+    const router = useRouter();
+    const currentPath = router.pathname;
+    const { navItems, navItemClass, navItemListClass, navClass, currentLinkClass = "" } = props;
+
+    return (
+        <nav className={navClass}>
```
Comment (Ilya-Valasiuk) [inline src/components/common/menus/NavList/NavList.tsx]:
```
<nav className={classNames(className,  "flex flex-row")}>
```

File: package.json (hunk 1 above)
Comment (Ilya-Pargoo) [reply package.json:20]: These packages are essential for TypeScript development in a React project, particularly when using Next.js. They provide type definitions for Node.js, React, and React DOM, which help with type checking and autocompletion in your IDE

I thought it's good idea to use ^ caret to update to patch and minor versions. Should I use it without it?

File: src/components/common/footer/Footer.tsx [hunk 29]
```diff
@@ -0,0 +1,68 @@
+import { FC } from "react";
+import classNames from "classnames";
+import Link from "next/link";
+import { NAV_ITEMS } from "./constants";
+import InstagramIcon from "@/../public/icons/instagram.svg";
```
Comment (Ilya-Valasiuk) [inline src/components/common/footer/Footer.tsx]: `@/../` ?

File: src/components/common/header/Header.tsx [hunk 30]
```diff
@@ -0,0 +1,62 @@
...
+                  )}>
+                  {item.text}
+                </Link>
+              </li>
+            ))}
+          </ul>
+        </nav >
+        <BurgerMenuButton onClick={toggleMobileMenu} isOpen={isMobileMenuOpen} />
+      </div>
+      {!!isMobileMenuOpen &&
```
Comment (Ilya-Valasiuk) [inline src/components/common/header/Header.tsx]: https://dev.to/ubahthebuilder/stop-conditional-rendering-in-react-without-knowing-this-and-ternary-operator-388

**Review by Ilya-Valasiuk — APPROVED** [review]

**Pull request merged**